svsim/README.md

```diff
@@ -0,0 +1,42 @@
+
+### Simulation
+
+The `Workspace`'s `compile` method returns a `Simulation` instance. A `Simulation` is launched using the `run` method, which passes a `Simulation.Controller` instance to the provided closure. The `Simulation.Controller` is used to explicitly control the `Simulation`, and should not escape the body of the closure. Like most of `svsim`, `Simulation.Controller` aims to be a low level API on which higher level APIs can be built. For example, `Simulation.Controller`'s `get` method does not implicitly run the simulation to make the effects of previous `set` calls visible, the way you would expect for a peek/poke test. To evaluate the simulation state, the user must explicitly call `controller.run(0)`. This, of course, does not prevent a peek/poke style API from being built using these building blocks in a higher-level framework.
+
+### Backend
+
+`svsim` currently provides two `Backend`s: `verilator.Backend` and `vcs.Backend`. Settings which **all** `svsim` `Backend`s must support (like SystemVerilog preprocessor defines) live in `SvsimCompilationSettings`, and each backend has its own backend-specific `CompilationSettings`.
+
+## Advanced Usage
+
+### `make simulation`
+
+Because `Backend`s are declarative, `svsim` is able to output a `Makefile` for compiling the simulation. This `Makefile` can be re-invoked after the fact to rebuild the simulation without re-running the Scala driver (`make simulation`). Users are free to modify either the harness generated by `svsim` (`generated-sources`), the SystemVerilog provided to `svsim` (`primary-sources`), or even the arguments passed to the compiler (`workdir-$tag/Makefile`) and these changes will be picked up when running `make simulation`.
+
+### `make replay`
+
+`svsim` by default emits an `execution-script.txt` when running a Simulation. This script captures all commands sent to the simulation, which enables `svsim` to add a second target to the `Makefile`: `make replay`. `make replay` rebuilds the simulation, picking up any changes as mentioned above, and then replays the commands that were sent by `Simulation.Controller` verbatim. This allows replaying most tests without having to re-run the Scala code, potentially with manual modifications to either the SystemVerilog or the test harness (though tests which modify their behavior based on values read from the `Simulation` may not be replayed faithfully).
```

Comment:
should we document the format of this replay file here? If not here, where is it documented? I see that it is not an ABI but it is a thing that if you're telling people they can edit we should at least point them to what the current format is.